File: Java/BinaryTrees/Medium/110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int dfsheight(TreeNode root){
        if(root == null)return 0;
        int leftHeight = dfsheight(root.left);
        if(leftHeight==-1)return -1;
        int rightHeight = dfsheight(root.right);
        if(rightHeight==-1)return -1;
        if(Math.abs(leftHeight-rightHeight)>1)return -1;
        return Math.max(leftHeight,rightHeight) +1;
    }
    public boolean isBalanced(TreeNode root) {
       return dfsheight(root) !=-1;
    }
}

File: Python/BinaryTrees/Easy/110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def depth(self, root):
        if root is None:
            return 0

        lh = self.depth(root.left)
        rh = self.depth(root.right)

        if (lh is -1 or rh is -1):
            return -1

        if abs(lh-rh) >1:
            return -1
        
        return 1+ max(lh,rh)
        
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if root == None:
            return True

        return True if self.depth(root) != -1 else False
        

        